### ISSUE TITLE
Can't put() more than once on SCPClient

### ISSUE THREAD

**Issue by 654sd** [jbardin/scp.py#196]
Are you using paramiko as a client or server?
Both

What feature(s) aren't working right?
SSH

What version(s) of paramiko are you using?
3.5.0

What version(s) of Python are you using?
3.8

What operating system and version are you using?
MacOS

If you're connecting as a client, which SSH server are you connecting to?
127.0.0.1

If you're using paramiko as part of another tool, which tool/version?
No response

Expected/desired behavior
I have encountered a strange problem, I use paramiko module ➕scp module to implement remote ssh upload file, but I only upload one file so that the code will execute successfully, the following is a successful example:

```
import paramiko

import scp

ssh = paramiko.SSHClient()

ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())

SSH. Connect (hostname = '127.0.0.1,

port=22,

username='root',

password='alpine')

pro = scp.SCPClient(ssh.get_transport())

ssh.exec_command('ls')

ssh.exec_command('cd / & ls -l')

pro.put('test', remote_path='/var', recursive=False)

print('put done')

pro.close()

ssh.close()
```

This code executes successfully and the console outputs the `put done` string

But the following code will cause an error:

```
import paramiko

import scp

ssh = paramiko.SSHClient()

ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())

SSH. Connect (hostname = '127.0.0.1,

port=22,

username='root',

password='alpine')

pro = scp.SCPClient(ssh.get_transport())

ssh.exec_command('ls')

ssh.exec_command('cd / & ls -l')

pro.put('test', remote_path='/var', recursive=False)

pro.put('test', remote_path='/var/1.txt', recursive=False)

print('put done')

pro.close()

ssh.close()
```

This code console error:

```
Traceback (most recent call last):

File "/Users/macos/PyCode/pythonProject/IOS/mina.py", line 14, in

pro.put('test', remote_path='/var/1.txt', recursive=False)

The File "/ Users/macos PyBin/T2Bypass/lib/python3.8 / site - packages/SCP. Py", line 198, put in

self.channel.exec_command(scp_command +

File "/ Users/macos PyBin T2Bypass/lib/python3.8 / site - packages/paramiko/channel. Py", line 70, in _check

return func(self, *args, **kwds)

File "/ Users/macos PyBin T2Bypass/lib/python3.8 / site - packages/paramiko/channel. Py", line 255, in exec_command

self._wait_for_event()

File "/ Users/macos PyBin T2Bypass/lib/python3.8 / site - packages/paramiko/channel. Py", line 1224, in _wait_for_event

raise e

paramiko.ssh_exception.SSHException: Channel closed.
```

So I find that the code works fine when I transfer the file only once, but when my code is uploaded two or more times my code will have an error paramiko.ssh_exception.SSHException: Channel closed. I tried to find a solution on goole, but it seems that no one met the same or similar situation with me, so I could not find a suitable solution. I suspect that this is a bug, because I made a judgment on the ssh connection status in the code that reported an error. It is found that the ssh connection status is still true when the upload code error is reported, which proves that the ssh ice has not broken the link. If you see the problem I raised, can you explain the reason for the error to me

Actual behavior
fix bug

How to reproduce
```
import paramiko
import scp

ssh = paramiko.SSHClient()
ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
ssh.connect(hostname='127.0.0.1',
port=22,
username='root',
password='alpine')
pro = scp.SCPClient(ssh.get_transport())
ssh.exec_command('ls')
ssh.exec_command('cd / & ls -l')
pro.put('test', remote_path='/var', recursive=False)
pro.put('test', remote_path='/var/1.txt', recursive=False)
print('put done')
pro.close()
ssh.close()
```

Anything else?
```
Traceback (most recent call last):
File "/Users/macos/PyCode/pythonProject/IOS/mina.py", line 14, in
pro.put('test', remote_path='/var/1.txt', recursive=False)
File "/Users/macos/PyBin/T2Bypass/lib/python3.8/site-packages/scp.py", line 198, in put
self.channel.exec_command(scp_command +
File "/Users/macos/PyBin/T2Bypass/lib/python3.8/site-packages/paramiko/channel.py", line 70, in _check
return func(self, *args, **kwds)
File "/Users/macos/PyBin/T2Bypass/lib/python3.8/site-packages/paramiko/channel.py", line 255, in exec_command
self._wait_for_event()
File "/Users/macos/PyBin/T2Bypass/lib/python3.8/site-packages/paramiko/channel.py", line 1224, in _wait_for_event
raise e
paramiko.ssh_exception.SSHException: Channel closed.
```

**Comment by remram44** [comment]
Please make an effort to put something meaningful in the title and format your message with correct markdown...

I'll look into this when I can.

**Comment by remram44** [comment]
I can't reproduce this. It works for me with openssh. Maybe there is a bug in your SSH server?

**Comment by 654sd** [comment]
> 我无法重现这个。它对我有用 openssh.也许您的 SSH 服务器中存在错误？

I suspected that there was a problem with my ssh service, but I used swifui provided by Xcode and CSharp under Windows to write, and found the same code and logic. I did not report any errors or exceptions in these two programming languages, so I ruled out the problem with ssh service. If you are unable to replicate this situation, I would like to try to establish contact with you and allow you to remotely control my computer to fix it, because I am really curious about the cause of this problem, I have looked at the source code of SCP Paramiko, but have not been able to find the problem, so I have been very wondering what the problem is, Moreover, this problem has been bothering me for nearly two weeks, and I really want to know the specific reason

**Comment by 654sd** [comment]
> 我无法重现这个。它对我有用 openssh.也许您的 SSH 服务器中存在错误？

I have doubted whether it was my SSH service that had the problem, but when I used Xcode to develop Swifui and wrote code in C# on Windows, I found that the same code and logic worked without any errors or exceptions in both programming languages. Therefore, I ruled out the possibility that the SSH service was the problem. If you cannot reproduce this situation, I would like to try to contact you and remotely control my computer to fix the issue, as I really want to know the specific cause of the problem. I have reviewed the source code of SCP Paramiko, but I couldn't find the problem. So, I have been very puzzled about where the issue lies, and it has bothered me for nearly 2 weeks. I really hope to know the exact cause of the problem.
